src/app/qgswmsdimensiondialog.cpp

```diff
@@ -0,0 +1,196 @@
+
+  // Set available names
+  for ( const QString &name : mServerProperties->predefinedWmsDimensionNames() )
+  {
+    if ( !alreadyDefinedDimensions.contains( name.toLower() ) )
+    {
+      mNameComboBox->addItem( name, QVariant( name.toLower() ) );
+    }
+  }
+
+  // Set default display to All values
+  mDefaultDisplayComboBox->setCurrentIndex( QgsVectorLayerServerProperties::AllDimensionValues );
+
+  mReferenceValueLabel->setEnabled( false );
+  mReferenceValueComboBox->setEnabled( false );
+}
+
+void QgsWmsDimensionDialog::setInfo( const QgsVectorLayerServerProperties::WmsDimensionInfo &info )
+{
+  int nameDataIndex = mNameComboBox->findData( QVariant( info.name ) );
+  if ( nameDataIndex == -1 )
+  {
+    mNameComboBox->setEditText( info.name );
+  }
+  else
+  {
+    mNameComboBox->setCurrentIndex( nameDataIndex );
+  }
+  mNameComboBox->setEnabled( false );
+
+  mFieldComboBox->setField( info.fieldName );
+  mEndFieldComboBox->setField( info.endFieldName );
+
+  mUnitsLineEdit->setText( info.units );
+  mUnitSymbolLineEdit->setText( info.unitSymbol );
+
+  mDefaultDisplayComboBox->setCurrentIndex( info.defaultDisplayType );
+  if ( info.defaultDisplayType == QgsVectorLayerServerProperties::ReferenceDimensionValue )
+  {
+    int referenceValueIndex = mReferenceValueComboBox->findData( info.referenceValue );
+    if ( referenceValueIndex == -1 )
+    {
+      mReferenceValueComboBox->setEditText( info.referenceValue.toString() );
+    }
+    else
+    {
+      mReferenceValueComboBox->setCurrentIndex( referenceValueIndex );
+    }
+  }
+  else
+  {
+    mReferenceValueComboBox->setCurrentIndex( 0 );
+  }
+}
+
+QgsVectorLayerServerProperties::WmsDimensionInfo QgsWmsDimensionDialog::info() const
+{
+  QString name = mNameComboBox->currentText();
+  if ( mNameComboBox->findText( name ) != -1 )
+  {
+    name = mNameComboBox->currentData().toString();
+  }
+
+  QString refText = mReferenceValueComboBox->currentText();
+  QVariant refValue;
+  if ( mNameComboBox->findText( name ) != -1 )
+  {
+    refValue = mReferenceValueComboBox->currentData();
+  }
+  return QgsVectorLayerServerProperties::WmsDimensionInfo( name, mFieldComboBox->currentField(),
+         mEndFieldComboBox->currentField(),
+         mUnitsLineEdit->text(), mUnitSymbolLineEdit->text(),
+         mDefaultDisplayComboBox->currentIndex(), refValue );
+}
+
+void QgsWmsDimensionDialog::nameChanged( const QString &name )
+{
+  // reset input
+  mUnitsLabel->setEnabled( true );
+  mUnitsLineEdit->setEnabled( true );
+  mUnitsLineEdit->clear();
+  mUnitSymbolLabel->setEnabled( true );
+  mUnitSymbolLineEdit->setEnabled( true );
+  mUnitSymbolLineEdit->clear();
+
+  const QStringList predefinedNames = mServerProperties->predefinedWmsDimensionNames();
+
+  if ( predefinedNames.contains( name ) )
+  {
+    if ( predefinedNames.indexOf( name ) == 0 )  // Time
```

Comment:
I think it would be safer (and cleaner ;)) to compare an enum rather than an index.